src/to_substrait.cpp

```diff
@@ -1313,6 +1315,43 @@ substrait::Rel *DuckDBToSubstrait::TransformGet(LogicalOperator &dop) {
 	return get_rel;
 }
 
+substrait::Expression_Literal DuckDBToSubstrait::ToExpressionLiteral(const substrait::Expression &expr) {
+	substrait::Expression_Literal literal_field;
+	switch (expr.rex_type_case())
+	{
+	case substrait::Expression::kLiteral:
+		literal_field = expr.literal();
+		break;
+	case substrait::Expression::kCast:
+		literal_field = ToExpressionLiteral(expr.cast().input());
```

Comment:
this looks wrong. why can we/should we ignore the cast. if it is right, let's add  a comment. (but I'm not clear how it could be write).

Reply:
I see your point. I think I did a mistake here.
We need to get literal from cast expression here. This is required because virtual table values are literal ([here](https://github.com/substrait-io/substrait/blob/1f3354d9f0f8b4425e98623f34d1e4578e2142bd/proto/substrait/algebra.proto#L91))

My understanding is, we need to convert this cast expression to literal based on duckdb type system. Let me know if this understanding is correct ?

Reply:
We need to handle it as Project<Expression>(Virtual table). This will be handled as separate PR